rotkehlchen/api/v1/schemas.py

```diff
@@ -943,14 +947,67 @@ def make_history_base_entry(
                 data: dict[str, Any],
                 **_kwargs: Any,
         ) -> dict[str, Any]:
-            return {'event': EthWithdrawalEvent(**data)}
+            return {'main_event': EthWithdrawalEvent(**data), 'related_events': None}
+
+    class CreateAssetMovementEventSchema(BaseSchema):
+        event_type = SerializableEnumField(
+            enum_class=HistoryEventType,
+            allow_only=[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL],
+            required=True,
+        )
+        fee = FeeField(load_default=None)
+        location = LocationField(required=True)
+        unique_id = fields.String(required=False, load_default=None)
+        event_identifier = fields.String(required=False, load_default=None)
+        asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
+        fee_asset = AssetField(load_default=None, required=False, expected_type=Asset, form_with_incomplete_data=True)  # noqa: E501
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            if (fee := data['fee']) is not None and data['fee_asset'] is None:
+                raise ValidationError(
+                    message='fee_asset must be provided if fee is set',
+                    field_name='fee',
+                )
+
+            if fee is not None and fee > ZERO:
+                main_event, *related_events = create_asset_movement_with_fee(
+                    fee=fee,
+                    asset=data['asset'],
+                    location=data['location'],
+                    timestamp=data['timestamp'],
+                    fee_asset=data['fee_asset'],
+                    event_type=data['event_type'],
+                    identifier=data.get('identifier'),
+                    amount=data['balance'].amount,
+                )
+            else:
+                main_event = AssetMovement(
+                    is_fee=False,
+                    asset=data['asset'],
+                    balance=data['balance'],
+                    location=data['location'],
+                    unique_id=data['unique_id'],
+                    timestamp=data['timestamp'],
+                    identifier=data.get('identifier'),
+                    event_type=data['event_type'],
+                    event_identifier=data['event_identifier'],
+                )
+                related_events = None
+
+            return {'main_event': main_event, 'related_events': related_events}
```

Comment:
I believe there is no reason to add two fields. Make it return a list and main_event be included inside the list